compiler-rt/lib/fuzzer/tests/CMakeLists.txt

```diff
@@ -63,10 +63,16 @@ if ("-fvisibility=hidden" IN_LIST LIBFUZZER_CFLAGS)
   list(APPEND LIBFUZZER_UNITTEST_CFLAGS "-fvisibility=hidden")
 endif()
 
-if(COMPILER_RT_DEFAULT_TARGET_ARCH IN_LIST FUZZER_SUPPORTED_ARCH)
-  # libFuzzer unit tests are only run on the host machine.
-  set(arch ${COMPILER_RT_DEFAULT_TARGET_ARCH})
+if(COMPILER_RT_CAN_EXECUTE_TESTS)
+  set(FUZZER_TEST_ARCH ${FUZZER_SUPPORTED_ARCH})
+  if (APPLE)
+    darwin_filter_host_archs(FUZZER_SUPPORTED_ARCH FUZZER_TEST_ARCH)
+  else()
+    # libFuzzer unit tests are only run on the host machine.
+    set(FUZZER_TEST_ARCH ${COMPILER_RT_DEFAULT_TARGET_ARCH})
```

Comment:
Note that this sets FUZZER_TEST_ARCH with just one arch, so the following loop will only run once, preserving the old behavior for non-Darwin targets (in Darwin it'll be a list). Perhaps you could extend the comment to point this out.